932 - Fix filter issues with datagrid (#965)
infor-design/enterprise-wc: src/components/ids-data-grid/README.md

- `columnGroups` {Array<object>} Allows you to group columns together in logical sets. See section below for details.
- `rowHeight` {string | `'xs'` | `'sm'` | `'md'` | `'lg'`} Sets the height of each row
- `data` {Array<object>} Sets the data to show in the data grid. This can be a JSON Array.

> - `disableClientFilter` {boolean} Sets to disable filter logic client side and let server do it.

Reviewer: ```suggestion
- `disableClientFilter` {boolean} Disables the filter logic client side in situations you want to filter server side.
```

> - `filterable` {boolean} Sets to turned on/off filters.

Reviewer: ```suggestion
- `filterable` {boolean} Turns on or off the filter functionality.
```

> - `filter-row-disabled` {boolean} Sets to disable filter row.

Reviewer: ```suggestion
- `filter-row-disabled` {boolean} Disables the filter row.
```

- `rowdeselected` Fires for each row that is deselected.
- `rowactivated` Fires for each row that is activated.
- `rowdeactivated` Fires for each row that is deactivated.

> - `clientfiltered` Fires after filter conditions changed, only fire with `disableClientFilter: true`.

Reviewer: ```suggestion
- `clientfiltered` Fires after filter conditions are changed. Only fires when `disableClientFilter: true`.
```

1. `UI Only` In order to do custom UI part of filter, add as html markup thru a slot. It must use slot and column-id attributes for example: `<div slot="filter-n" column-id="n">...</div>` where n is the columnId same passed in the columns.
1. `filterFunction` This is a user defined filter method which must return a boolean. It determines if a cell value should be considered as a valid filtered value.

> 1. `disableClientFilter` This is an api setting to disable filter logic client side. It will set filter conditions and fire an event `clientfiltered` which can listen for custom logic.

Reviewer: Any reason to not use the same filtered event?

Author: updated event name as `filtered`
